types/react-navigation/index.d.ts

```diff
@@ -86,7 +87,7 @@ export type NavigationLeafRoute<Params> = {
    * Params passed to this route when navigating to it,
    * e.g. `{ car_id: 123 }` in a route that displays a car.
    */
-  params?: Params,
+  params: Params,
```

Comment:
I totally agree with "we probably want it to be as accurate as possible". If a screen expects its params to be some `T`, it should be free to express this, other than be forced to be `T | undefined`. As in typical react component situation, a component could expect its props type `P`. If an undefined is expected, it could be expressed as `T | undefined` explicitly. Although currently, this could not be enforced by the compiler as props do. I think it would be a logical error to provide an unexpected undefined (as params). To force the (undefined) check does not solve the problem, and make the boundary fuzzy.